dashboard/app/dsl/script_dsl.rb

```diff
@@ -76,21 +78,43 @@ def level_concept_difficulty(json)
 
   string :video_key_for_next_level
 
+  boolean :active
+
   def assessment(name)
     level(name, {assessment: true})
   end
 
   def level(name, properties = {})
-    @levels << {
+    level = {
       :name => name,
-      :stage => @stage,
       :stage_flex_category => @stage_flex_category,
       :skin => @skin,
       :concepts => @concepts.join(','),
       :level_concept_difficulty => @level_concept_difficulty || {},
       :video_key => @video_key_for_next_level
     }.merge(properties).select{|_, v| v.present? }
     @video_key_for_next_level = nil
+    if @current_scriptlevel
+      @current_scriptlevel[:levels] << level
+      if !@active
+        @current_scriptlevel[:properties][name] = { :active => @active }
+        @active = true
+      end
+    else
+      @scriptlevels << {
+        :stage => @stage,
+        :levels => [level]
+      }
+    end
+  end
+
+  def variants
+    @current_scriptlevel = { :levels => [], :properties => {}, :stage => @stage}
+  end
+
+  def endvariants
+    @scriptlevels << @current_scriptlevel
+    @current_scriptlevel = nil
```

Comment:
I love that proposal for a block-based format. Is the issue that we would have to add support for block-based methods in general, or simply that it would be nice to have consistency?